test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -691,4 +691,16 @@ public final void testIndexTimeStoredFieldsAccess() throws IOException {
             .fielddataBuilder("test", lookupSource)
             .build(new IndexFieldDataCache.None(), new NoneCircuitBreakerService());
     }
+
+    public final void testNullInput() throws Exception {
+        assumeTrue("Mapper does not permit null values", allowsNullValues());
```

Comment:
should we actually check that when the mapper does not allow null values, an exception is thrown?

Reply:
that's a good idea.  I think I'm going to revert the changes to rank_features and dense_vectors as well and add them as TODOs because I want to get this into 7.13 and they should get some more discussion.